### PR TITLE
Fix #518

### LINKED ISSUE #518: Consult register actions (M-p, M-w, ...) do not work in terminal
Hi:

Recently I found that Modified keys are not working in consult properly when using emacs form the terminal (-nw).

Commands like `consult-register-save` receive `w` instead of `M-w`, `f` instead of `M-f` and so on. Before I had ivy and it worked properly; and xterm is sending the right bindings because the rest of emacs gets them.  And when I do `M-x C-h k M-w` it shows the help for `M-w`... so this seem to be something specific in Consult??? BTW I use `Vertico` in case you need to know.

Thanks in advance, Ergus

### PR DESCRIPTION
**Pull request by oantolin**
In terminal Emacs, given M-char, (read-key) will return 27 and set last-input-event to char. This change turns that situation into what would have been read in graphical Emacs, letting the rest of the code remain unchanged.

### CODE REVIEW

File: consult-register.el [hunk 1]
```diff
@@ -252,10 +253,15 @@ This function is derived from `register-read-with-preview'."
...
-                                             'face 'minibuffer-prompt))
+	  (while (memq (setq key
+                             (read-key (propertize (caddr (assq action action-list))
+                                                   'face 'minibuffer-prompt)))
 		       help-chars)
             (funcall preview))
+          (when (and (eql key ?\e) (characterp last-input-event))
+            ;; in terminal Emacs M-letter is read as two keys, ESC and the letter,
+            ;; turn that into what would have been read in graphical Emacs
+            (setq last-input-event (logior #x8000000 last-input-event)))
```
Comment (minad) [inline consult-register.el:264]: I am not comfortable with overwriting a variable which I don't own. Please introduce a separate variable here for the key decoding.

**Comment by minad** [comment]
Thanks for looking into this!

File: consult-register.el (hunk 1 above)
Comment (oantolin) [reply consult-register.el:264]: Will do.

Comment (oantolin) [reply consult-register.el]: It looks better with the new variable anyway.

Comment (oantolin) [reply consult-register.el]: I forgot to reindent, wait a second.

Comment (oantolin) [reply consult-register.el]: OK, should be good now.

**Pull request merged**